### ISSUE TITLE
🐛 noUnusedVariables isn't considering local usage of type

### ISSUE THREAD

**Issue by matthewlilley** [biomejs/biome#546]
### Environment information

```block
CLI:
  Version:                      1.2.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.18.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.9.2"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0

Discovering running Biome servers...

Incompatible Biome Server: ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

ℹ Rage discovered this running server using an incompatible version of Biome.

Server:
  Version:                      1.1.2

Running Biome Server: ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

ℹ The client isn't connected to any server but rage discovered this running Biome server.

Server:
  Version:                      1.2.2
  Name:                         biome_lsp
  CPU Architecture:             x86_64
  OS:                           linux

Workspace:
  Open Documents:               0

Other Active Server Workspaces:

Workspace:
  Open Documents:               2
  Client Name:                  Visual Studio Code
  Client Version:               1.83.1

Biome Server Log:

⚠ Please review the content of the log file before sharing it publicly as it may contain sensitive information:
  * Path names that may reveal your name, a project name, or the name of your employer.
  * Source code

├─┐biome_lsp::server::initialize{capabilities=ClientCapabilities { workspace: None, text_document: None, window: None, general: None, experimental: None }, client_info=ClientInfo { name: "biome_service", version: Some("1.2.2") }}
│ ├─0ms INFO biome_lsp::server Starting Biome Language Server...
├─┘
├─┐biome_lsp::server::biome/rage{params=RageParams}
├─┘
```

### What happened?

noUnusedVariables isn't considering local usage

![image](https://github.com/biomejs/biome/assets/6063350/5f6aa5a5-b05b-47b0-8bb7-d5e6074d6648)

### Expected result

This shouldn't flag as a lint error

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
I believe this is already fixed. Can you check the [playground](https://biomejs.dev/playground/)? We'll soon make a release

**Comment by Conaclos** [comment]
This should also be solved with the new 1.3 release.

**Comment by matthewlilley** [comment]
@ematipico @Conaclos Tried the playground, even when types aren't used it doesn't appear to say anything about that.

https://biomejs.dev/playground/?code=dAB5AHAAZQAgAFQAZQBzAHQAIAA9ACAAYgBvAG8AbAAKAAoAZgB1AG4AYwB0AGkAbwBuACAAdABlAHMAdAAoAHQAOgAgAGIAbwBvAGwAKQAgAHsAfQA%3D

Might be missing something? First time using the biome playground but seems linter is enabled.

**Comment by Levieber** [comment]
> Might be missing something? First time using the biome playground but seems linter is enabled.

You have to enable the noUnusedVariables rule, as I remember it is not enabled by default.

![Select lint rules to apply in playground](https://github.com/biomejs/biome/assets/104157600/2092613b-d7fb-42cd-880a-e856b739b9c0)

The way I found is to enable all rules.

**Comment by matthewlilley** [comment]
Ah yeah, got it. Seems to be working as expected here.

**Comment by matthewlilley** [comment]
Oh, you already published 1.3.0 as well. Thanks!
